gms/gossiper.cc

```diff
@@ -252,13 +252,23 @@ future<> gossiper::do_send_ack_msg(msg_addr from, gossip_digest_syn syn_msg) {
 
 static bool should_count_as_msg_processing(const std::map<inet_address, endpoint_state>& map) {
     bool count_as_msg_processing  = false;
-    for (auto& x : map) {
-        auto& state = x.second;
+    for (const auto& x : map) {
+        const auto& state = x.second;
         for (const auto& entry : state.get_application_state_map()) {
-            auto& app_state = entry.first;
-            if (!(app_state == application_state::LOAD ||
-                  app_state == application_state::VIEW_BACKLOG ||
-                  app_state == application_state::CACHE_HITRATES)) {
+            const auto& app_state = entry.first;
+
+            auto is_critical_state = [](application_state state) {
```

Comment:
TBH I liked the previous version more. This is pretty subjective

Reply:
Yeah, ... it's that I was adding another condition, so the long `if` statement starts looking unwieldy (also the negation made the clang-tidy nervous and offered switching the condition equalities and ands/ors). This way it might also be more clear what we are checking here (although we might think about a better name than `is_critical_state`)